packages/react-components/src/components/EndCallButton.tsx

```diff
@@ -63,10 +62,16 @@ export const EndCallButton = (props: EndCallButtonProps): JSX.Element => {
     styles ?? {}
   );
 
+  const onHangUp = (): void => {
+    if (props.onHangUp) {
+      props.onHangUp();
+    }
+  };
+
   return (
     <ControlBarButton
       {...props}
-      onClick={props.onHangUp ?? props.onClick}
+      onClick={props.onHangUp ? onHangUp : props.onClick}
```

Comment:
we are checking `props.onHangUp` twice, once here and once in `onHangUp`. Can changes to this file be reverted? (is there any change in behavior in this file?)

Reply:
in onHangup is checking and calling onHangup, in here props.onHangUp ? onHangUp : props.onClick is if props.onHangUp is not passed in then don't use onHangup function, use onClick instead

Reply:
No changes in behaviour, previously it's onClick = {props.onHangup?? props.onClick}, however now onHangup has an optional prop so we can't assign it straight to onCLick anymore, need to do something like onClick = () =>{ props.onHangup()} instead

Reply:
Something to look at here is currently, using this component, how would someone use onHangUp here to call `forEveryone` as `true`? This currently always calls `props.onHangUp()` with no parameter.

Instead you'll want this pure component to always have onHangUp be a simple function that is just `onHangUp?: () => Promise<void>;`

It is then up to the parent of the pure EndCallButton component to decide how the hang up function is implemented, and if it kicks everyone out of the call or not`

Reply:
The challenge here is onhangeup is grabbed by usepropsfor so even if I set it up as onHangUp?: () => Promise<void>, what's passed down to endcallButton is still going to be onHangUp?: (foreveryone) => Promise<void>;